### ISSUE TITLE
ST10: Regression with 3.3.0, causes false positives with named parameters

### ISSUE THREAD

**Issue by paulcwatts** [sqlfluff/sqlfluff#6493]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

After upgrading from sqlfluff 3.2.0 to 3.3.0, existing queries are now reporting a redundant constant expression. However, it seems to caused by a placeholder parameter that is the same name as the variable on the left hand side of the equality. 

I can reproduce the error with a simple query. Create a file called `test.sql`

```sql
SELECT a FROM b
WHERE a = :a
```

Then run the command: `sqlfluff lint --dialect=postgres test.sql`

```
== [test.sql] FAIL                                                                                                                                                                                      
L:   2 | P:   9 | ST10 | Redundant constant expression.
                       | [structure.constant_expression]
```

If you change the name of the named parameter to something like `:ab`, the error goes away.

I don't know if this is related to https://github.com/sqlfluff/sqlfluff/issues/6492, but they are regressions related to the same version and same rule.

### Expected Behaviour

I don't think this should be considered a redundant constant expression, since the right hand side is not constant.

### Observed Behaviour

See above.

### How to reproduce

See the sql and command above, with the configuration below.

### Dialect

postgres

### Version

sqlfluff, version 3.3.0

### Configuration

[sqlfluff]
templater = placeholder

[sqlfluff:templater:placeholder]
param_style = colon
a = 'a'

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by rogalski** [comment]
Thanks.

I probably just need to exclude templated expressions from check.
I'll provide a PR with fix ASAP.
